### ISSUE TITLE
[pickers] DateRangePicker@5 gives deprecation warning for using `startText`, `endText` but non-deprecated usage does not work. 

### ISSUE THREAD

**Issue by dwjohnston** [mui/mui-x#10390]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

V5
https://codesandbox.io/s/cocky-colden-nwn6dl?file=/demo.tsx

In this sandbox there are two instances of the DateRangerPicker

The first uses `startText` and `endText` and we can see that this works. However, in the console note a deprecation warning, and the link there tells us to the use `localeText`. 

The second uses localeText but we can see that it does not override the values provided by the LocalizationProvider. 

This shows a type error suggesting the `localeText` is not a valid property the DateRangerPicker. 

For posterity, here is the V6 sandbox: 
https://codesandbox.io/s/cool-knuth-k54ghs?file=/Demo.tsx

Here we can see that using `localeText` does override the LocalizationProvider values. 

### Current behavior 😯

The `localeText` property does not override the values given by the LocalizationProvider. 

### Expected behavior 🤔

The `localeText` property should override the values given by the LocalizationProvider. 

### Context 🔦

Cleaning up browser warnings. I have to keep using the `startText` `endText` solution, which means the deprecation warning remains. 

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by LukasTy** [comment]
> In this sandbox there are two instances of the DateRangerPicker
> 
> The first uses `startText` and `endText` and we can see that this works. However, in the console note a deprecation warning, and the link there tells us to the use `localeText`.

Hello, could you please double-check your example?
It does seem to contain only a single instance that uses `localeText` on `LocalizationProvider` and works without problems.
Specifying `startText` and/or `endText` props directly on the `DateRangePicker` also works and takes precedence, but, as mentioned, throws a deprecation message.

From what I see, I struggle to understand the problem if there is any. 🤷

**Comment by dwjohnston** [comment]
@LukasTy Thanks, I've updated the link.

**Comment by LukasTy** [comment]
> Also, where I might have been going wrong is seeing the deprecation warning, following the link, and then visiting the docs, but seeing the V6 docs and erroneously trying the `localeText`solution.

Yes, that could have been the root problem that caused the confusion.
Pickers on v5 do not support the `localeText` prop, which has been only added in v6.
Opening your codesandbox marks the `localeText` on `DateRangePicker` as an unknown prop.

> In any case - there isn't a 'local overide' option for the components - I guess I could wrap it in it's own LocalizationProvider?

Yes, that would be the idea. You can either wrap the second `DateRangePicker` in a separate `LocalizationProvider` or simply change the `localeText` prop values on the root `LocalizationProvider` as it won't affect the first instance, which has `startText` and `endText` props specified directly. 👍 

TL/DR: The feature seems to be working as expected.
If you think that there still is an issue - please feel free to re-open the issue or create a new one. 😉
